pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -453,56 +451,48 @@ func updateEmbeddedExpr(expr parser.Expr, call *parser.Call) parser.Expr {
 	}
 }
 
-// updateRangeInterval returns true if range interval was updated successfully,
-// false otherwise
-func updateRangeInterval(expr parser.Expr, rangeInterval time.Duration) bool {
-	switch e := expr.(type) {
-	case *parser.AggregateExpr:
-		return updateRangeInterval(e.Expr, rangeInterval)
-	case *parser.Call:
-		var updated bool
-		// Iterate over Call's arguments until a MatrixSelector is found
-		for _, arg := range e.Args {
-			updated = updateRangeInterval(arg, rangeInterval)
-			if updated {
-				break
-			}
+// updateRangeInterval modifies the input expr in-place and updates the range interval on the matrix selector.
+// Returns an error if 0 or 2+ matrix selectors are found.
+func updateRangeInterval(expr parser.Expr, rangeInterval time.Duration) error {
```

Comment:
I think this cannot happen, but should we check if `rangeInterval` < 0?

Reply:
It shouldn't but better check it!